### ISSUE TITLE
Mix of SSL and non-SSL listeners

### ISSUE THREAD

**Issue by jasonmiraglia** [kubernetes-sigs/aws-load-balancer-controller#1729]
I'm looking for some guidance on how to configure a load balancer that requires a mix of security policies. We can use the following annotations to control the policy and cert for **all** listeners
```
service.beta.kubernetes.io/aws-load-balancer-ssl-cert
service.beta.kubernetes.io/aws-load-balancer-ssl-negotiation-policy
```

which is fine for configuring an NLB like: 
TLS: 1000 | Security Policy: ELBSecurityPolicy-TLS-1-2-Ext-2018-06 | SSL Certificate: ...
TLS: 1001 | Security Policy: ELBSecurityPolicy-TLS-1-2-Ext-2018-06 | SSL Certificate: ...
TLS: 1100 | Security Policy: ELBSecurityPolicy-TLS-1-2-Ext-2018-06 | SSL Certificate: ...

But, this won't work for configuring an NLB like:
Listener: 
TLS: 1000 | Security Policy: ELBSecurityPolicy-TLS-1-2-Ext-2018-06 | SSL Certificate: ...
TLS: 1001 | Security Policy: ELBSecurityPolicy-TLS-1-2-Ext-2018-06 | SSL Certificate: ...
TCP: 1100 | Security Policy: N/A | SSL_Certificate: N/A

Is anyone able to shed some light on how one might go about achieving this?

**Comment by kishorj** [comment]
You can use the `service.beta.kubernetes.io/aws-load-balancer-ssl-ports` annotation to mix tls and non-tls listeners. But the certs and security policies will be the same for all TLS listeners for the given NLB.

Here is a possible solutoin -
You can use the `TargetGroupBinding` resources to bind to each of the externally managed target group on your NLB. 
- Create NLB and add the listeners and target groups externally
- Create a k8s service of type ClusterIP (without any of the aws annotatonis)
- Create targetgroup bindings for your externally managed target groups, and refer to the service created above to associate with the backend pods

I haven't tried it myself, and we don't have further documentation on this.

**Comment by jasonmiraglia** [comment]
@kishorj Thanks so much for the quick response. `service.beta.kubernetes.io/aws-load-balancer-ssl-ports` seems to be exactly what I was looking for here. However, it's producing some unexpected results.

```
annotations:
  service.beta.kubernetes.io/aws-load-balancer-backend-protocol=tcp
  service.beta.kubernetes.io/aws-load-balancer-ssl-cert=SOME_CERT
  service.beta.kubernetes.io/aws-load-balancer-ssl-negotiation-policy=ELBSecurityPolicy-TLS-1-2-Ext-2018-06
  service.beta.kubernetes.io/aws-load-balancer-ssl-ports=443,4109,4009,4110,4010,4011

ports:
  443:31786/TCP
  4109:32610/TCP
  4209:32269/TCP
  4009:30796/TCP
  4110:31697/TCP
  4010:32415/TCP
```

Results in an error creating a listener for port 4209:
```
{
   "level":"error",
   "ts":1608049505.2679331,
   "logger":"controller",
   "msg":"Reconciler error",
   "controller":"service",
   "name":"oc-site-gateway",
   "namespace":"oc-app",
   "error":"ValidationError: An SSL policy cannot be specifed for TCP listeners\n\tstatus code: 400, request id: 10c838fb-634c-4121-ab75-363f412a1931"
}
```

Seems that the `service.beta.kubernetes.io/aws-load-balancer-ssl-negotiation-policy` can maybe not be used in conjunction with `service.beta.kubernetes.io/aws-load-balancer-ssl-ports`

**Comment by kishorj** [comment]
The reconciler error is due to issue #1661. It has been fixed in 2.1.0, your service spec should work as expected after you upgrade the controller.

This annotation is not required in your case - `service.beta.kubernetes.io/aws-load-balancer-backend-protocol=tcp`.

**Comment by jasonmiraglia** [comment]
You rock @kishorj . Upgrading to `2.1.0` totally did the trick and now working as expected. Thanks so much for the assistance on this.
